# BCB546-Spring 2022

## Author-year markdown file for Ice Cream Bean group.

Title: _Genome-Wide Comparative Analysis of Flowering-Related Genes in Arabidopsis, Wheat, and Barley_

Authors: Fred Y. Peng,1 Zhiqiu Hu,2 and Rong-Cai Yang, 2015.

Published in International Journal of Plant Genomics.

The authors examined flowering genes two different plant species; wheat (Triticum aestivum L.) and barley (Hordeum vulgare L.)  Because the genomes of these two species are large and complicated, identification of specific flowering genes was difficult. In order to predict these  genes, the authors used known flowering genes from Arabidopsis (Arabidopsis thaliana). In total, they identified 900 and 275 flowering genes in wheat and barley. These putative genes were clustered into 144 ortholog groups and showed that orthologous gene pairs in three flowering gene families; PEBP, MADS, and BBX. Now that these flowering genes have been identified, they can be used in breeding to induce early flowering wheat and barley, increasing grain yield and quality in these two species.

# Figure Recreation
## Table 2
  Table 2 presented the new annotation percentages of characterized and uncharacterized flowering related genes of barley and wheat based on BLAST hits of Arabidopsis flowering genes. To replicate this data, Supplementary file 3 was used and sorted to view 'uncharacterized' and 'characterized' proteins. This was done using the group.by function and sorting for 'Unnamed: 1' column in python. A total of 866 uncharacterized wheat proteins were found along with 34 characterized proteins. Their percentages were calculated based on the total amount of proteins for each species. For Barley, data was also collected from Supplementary file 3 and sorted for 'uncharacterized' and 'characterized' proteins in the same way as mentioned above. A total of 259 'uncharacterized' proteins were found and '16' characterized proteins were calculated. These results agree with the authors findings for Table 2. 
## Table 3
  Table 3 examined the structural characteristics of flowering genes in Arabidopsis, wheat and barley. Multiple characteristics were explored, including the number of transcripts per gene, gene length, exons per gene, exon size, intron size, and protein length. All of these structural characteristics were replicated in using python and Supplemental file 4. First, flowering genes from Arabidopsis, wheat and barley were sorted in python and sorted using the first column to find the number of transcripts per gene (Unnamed:1 column). For all three species, the replicated data agreed with author data. Next, sorted exon data was replicated. The file was sorted according to average number of exons (Unnamed: 3 column). Exon size was calculated by first creating a copy of the file and dropping the header to allow the data to be changed to numeric values using pandas. Then the average, minimum, and maximum values were calculated. The data gathered for this characteristic agreed with the authors. For intron size, the same analysis was carried out for average and maximum values. 
  
  In order to calculate minimum values, the row had to be sorted to account for genes that contained no introns (to exclude them from the range). The data for intron size for all three also agreed with the authors. For gene length, a new column was made for average number of exon multipled by exon length for each gene present(titled exon_total). Then, another new column was created to calculate gene length as a whole. The second new column, titled exon_intron, was the addition of exon_total column + average number of introns to obtain total gene length. The average, minimum, and maximum for each species was calculated. The results from these calculations did not match the authors. In the manuscript, there was not an explanation as to how this value was calculated. Protein length was calculated by taking the exon_total column and calculating the average, minimum, and maximum values. Again, this number differed from the authors' values. The differences in these last two characteristics could be due to the fact that some data points were excluded in the authors' results, but this was not explainined in the manuscript. Supplemental file 4 contained more data points than was listed on the table above each species.

*Table creation in Python:*
For both tables, prettytable on python was used. Each row was added using the add_row function and exported using .get_string(). Txt.files of each table can be seen the results folder under the respective tab for each table.

*Table update in R:*
To try to make the tables match more closely to the authors', R was used. Using the kableExtra package in R, the tables were able to resemble the authors' more closely, but are still not similar. The kable_paper(bookstrap_options) function was used to clean the tables up and they were exported from R as PNG files. They can be viewed under the results tab of our repository under their respective folders. 

## Figure 1
  Figure one is a multiple sequence alignment of the OG5_178217 orthogroup where Figure 1a is conserved MADS-box domain and K-box region. Blue is the MADS-box domain and red is K-box region. Figure 1b is the PEBP family proteins. FASTA files for each of the genes within the alignment for arabidopsis, wheat, and barley were retrieved from the repository using Git Bash. In figure 1a, it was unclear which iteration of the MLOC gene was used, so all three were included. In Figure 1b,Traes_3B_2A454DB62 was unavailable from all major databases. 
  
  To create figure specific files using BioPython, a text file of gene lists for both figures were created. Then, all the separate species files were compiled into a single file as a dictionary. The gene lists iterated over the species sequence and pulled out Fig_1a and Fig_2b sequences into seaprate files. The authors compiled their sequences from targeting orthologous flowering genes using BLAST, OrthoMCL-DB, and IntProScan 5, but that would have produced more data than we needed or could computationally handle, so just the genes from the figure were compiled.  
  
  For sequence alignment, Clustalx v2.0.12 was installed and sequences for each figure uploaded. The authors did not indicate any parameters, quality scores, or trimming. A visual inspection showed the sequence lengths and alignment were similar to the study. While the order of the species' sequences were different, changing the parameters did not seem to make a big difference. The output were nexus files for phylogenetic analysis, as indicated by the authors. Jalview v2.0 was not available, so 2.11.2.1 was downloaded. Again, no parameters or quality scores were given by the authors, so standard parameters were kept. The "colour" option to change to "Clustalx" refelcted the box domains described in the figure. The quality scores provided by Jalview were really low towards the end of the sequence, as expected by deblocking/noise, but did not seem to be trimmed by the authors as the sequence lengths were the same. These alignments were then output as image files and cropped to fit the page as the study's figure was. 
