# Code
This directory contains the code written to reproduce figures and tables in Peng et al. 2015.

## Contents of this directory
This directory contains: 
1. `Fig_1_code`: Jupyter notebook detailing the reproduction of Figure 1, multiple sequence alignment of OG5_178217, excluding the sequence retrieval steps.
2. `Fig_1_sequenceretrieval`: Markdown file detailing how to retrieve sequences used to make figure one using HPC-class. 
3. `Figure_2`: Markdown file detailing the initial steps to replicate Figure 2, phylogeny of PEBP and MADS family proteins in Arabidopsis, wheat, and barley, including using GUI applications such as Beauti.
4. `Figure_2`: R markdown file describing finishing steps for replicating Figure 2 in R. 
5. `Figure 3`: R markdown file describing steps used to replicate Figure 3, heat maps of expression of flowering genes in different tissues and development stages in Arabidopsis, wheat, and barley. 
6. `Figure_4_and_S3`: R markdown file desribing steps used to replicate Figure 4 and Supplemental Figure S3, line graphs of expression of orthologous genes in different tissues and development stages in wheat and barley.
7. `Supp_Data_2`: Markdown file detailing the steps to replicate `Supplemental data 2`, ortholog group analysis. 
8. `Table_1_and_S1`: R markdown file detailing how to replicate Table 1 and Supplemental Table S1,  distributions of flowering genes over chromosomes and functional groups in Arabidopsis.
9. `Table_2_Python_Code`: Jupyter notebook detailing replication of Table 2, flowering-related genes in barley and wheat annotated by top BLAST hits in *Arabidopsis thaliana*.
10. `Table_3_Python_Code`: Jupyter notebook detailing replication of Table 3, structural characteristics of flowering-related genes in Arabidopsis, wheat, and barley.
11. `UpdatingTables2and3RMDfile`: R markdown file detailing steps to replicate Tables 2 and 3 in R to improve aesthetics. 
12. .Rhistory: R History Source file.